conans/client/tools/scm.py

```diff
@@ -89,23 +99,52 @@ def run(self, command):
         command = self._configure_ssl_verify + command
         return super(Git, self).run(command)
 
-    def clone(self, url, branch=None, args=""):
+    def _fetch(self, url, branch, shallow):
+        if not branch:
+            raise ConanException("The destination folder '%s' is not empty, "
+                                 "specify a branch to checkout (not a tag or commit) "
+                                 "or specify a 'subfolder' "
+                                 "attribute in the 'scm'" % self.folder)
+
+        output = self.run("init")
+        output += self.run('remote add origin "%s"' % url)
+        if shallow:
+            try:
+                output += self.run('fetch --depth 1 origin "%s"' % branch)
+                output += self.run('checkout FETCH_HEAD')
+            except CalledProcessError as e:
+                rmdir(os.path.join(self.folder, ".git"))
```

Comment:
If we accept that this code is also run in a local folder, I think that we cannot remove this local `.git` folder even if something goes really wrong.

Reply:
yes, probably, lemme try to rerun tests without this one, it was needed to do a fallback in case of shallow-clone failed

Reply:
I've moved fallback rmdir to the `conan\conans\model\scm.py` with explanation why it should be there